src/Gui/CommandActionPy.cpp

```diff
@@ -0,0 +1,151 @@
+/***************************************************************************
+ *   Copyright (c) 2023 Mario Passaglia <mpassaglia[at]cbc.uba.ar>         *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 51 Franklin Street,      *
+ *   Fifth Floor, Boston, MA  02110-1301, USA                              *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "Command.h"
+#include "Action.h"
+#include "PythonWrapper.h"
+
+#include "CommandActionPy.h"
+#include "CommandPy.h"
+
+
+using namespace Gui;
+
+CommandActionPy::CommandActionPy(Py::PythonClassInstance* self, Py::Tuple& args, Py::Dict& kwds)
+    : Py::PythonClass<CommandActionPy>::PythonClass(self, args, kwds)
+{
+    const char* name;
+    if (!PyArg_ParseTuple(args.ptr(), "s", &name)) {
+        throw Py::Exception();
+    }
+
+    cmdName = name;
+    cmd = Application::Instance->commandManager().getCommandByName(name);
+}
+
+CommandActionPy::~CommandActionPy()
+{
+}
+
+Py::Object CommandActionPy::getAction()
+{
+    if (!cmd) {
+        cmd = Application::Instance->commandManager().getCommandByName(cmdName.c_str());
```

Comment:
Hasn't this already been run on line 42? Will there ever be a case where `cmd` hasn't been set during construction?

Reply:
The object can be created prior to the existence of the command. In that case `getCommandByName` returns `nullptr`.
Therefore, to obtain the action in subsequent uses of the descriptor, the existence of the command must be checked.